tests/common/test_parser.py

```diff
@@ -0,0 +1,24 @@
+import sys
+import os
+import subprocess
+
+sys.path.append("..")
+
+
+def test_create_parser():
+    ip = "10.19.111.54"
+    user_token = "qBIErMvrucSzVhwY92rC6bMq7Tj22N0nlCmhAxaO"
+    os.environ["GLPI_INSTANCE"] = ip
+    os.environ["GLPI_TOKEN"] = user_token
+    output = subprocess.check_output(["./parser.py"], cwd="../../common")
+    out = output.split()
+    assert ip == out[0].decode("ascii")
+    assert user_token == out[1].decode("ascii")
+    os.environ["GLPI_INSTANCE"] = "1"
+    os.environ["GLPI_TOKEN"] = "2"
+    output = subprocess.check_output(
+        ["./parser.py", "-i", ip, "-t", user_token], cwd="../../common"
+    )
+    out = output.split()
```

Comment:
Sorry, my bad - I was referring to the strategy of testing the class by calling it directly, rather than passing it in via `subprocess`. So something like this:
```
from common.parser import argparser

def test_create_parser():


    ip = "127.0.0.1"
    user_token = "test"
    os.environ["GLPI_INSTANCE"] = ip
    os.environ["GLPI_TOKEN"] = user_token
    parsers = argparser()
    args = parsers.parser.parse_args()
    assert ip == args.ip
    assert user_token == args.token
```

You could also leave the test in `tests/common/test_parser.py` - would make it easier if/when we add `pytest` to ci/cd workflow.

@dkosteck what do you think?

Reply:
I agree with this generally, and that it's the right approach. Eric and I discussed this before the implementation, but I think the issue would become when testing the passed in arguments, we need a way to set the argparse arguments programmatically instead of via CLI (--ip flag, --token, set via Python, etc.), not necessarily for testing the environment variables but the other permutations. I mentioned to Eric that if we did not have a good way to do this at the moment then we could come back to it, but I think adding it at least for just the environment case right now, as @adhil0 outlined it makes sense @ericz2002.